### ISSUE TITLE
Imported User Authentication Cert Not Remembered

### ISSUE THREAD

**Issue by davewichers** [zaproxy/zaproxy#2489]
I'm using the Options-->Certificate feature:

I import my cert, and then go to the KeyStore and select my cert and Set Active on that cert. And it works. That's great.

But then I notice at the top of the dialog it says: "Certificate setting will not be stored in options and you will need to enable certificate next time you restart ZAP."

That kind of sucks. Why is that? Can't it pretty easily be remembered? If you are worried about storing the cert password, then after restarting, the first time the cert needs to be used, you could prompt the user for the cert password again. That would be much easier than having to import it again, and then set that cert to be the active cert.

Also - if I want to fully automate the use of ZAP in regression tests, how would I save the cert so it can be used automatically and repeatably? Maybe allow a properties entry to be set in some file somewhere, where the cert password can be saved? Actually, I think the simplest thing to do that, but also via the GUI when you enter the cert password, have a checkbox that says "Save this password for this cert" or something like that, so you could easily set it via that GUI and wouldn't have to hack around in any files.

**Comment by kingthorin** [comment]
To your second point there's a ticket open: https://github.com/zaproxy/zaproxy/issues/2023

**Comment by kingthorin** [comment]
Client certs can now be set via the cli and api so I think this is kind of a non-issue?

**Comment by davewichers** [comment]
I disagree. ZAP should support the scenario I originally described. I.e., I open ZAP, import a cert. via the ZAP UI, and it should 'remember' the cert I imported. I shouldn't be forced to use the CLI to load it each time.

**Comment by kingthorin** [comment]
That’s fair. However, given the minimal group that uses client certs and even more minimal group for which this is an issue I don’t see it being addressed. Like bordering on `wontfix` (for my 2 cents).
